atom/browser/native_window.cc

```diff
@@ -805,6 +810,25 @@ void NativeWindow::ScheduleUnresponsiveEvent(int ms) {
       base::TimeDelta::FromMilliseconds(ms));
 }
 
+void NativeWindow::SetHtmlApiFullscreen(bool enter_fullscreen) {
+  // Window is already in fullscreen mode, save the state.
+  if (enter_fullscreen && IsFullscreen()) {
+    forced_fullscreen_ = true;
+    html_fullscreen_ = true;
+    return;
+  }
+
+  // Exit html fullscreen state but not window's fullscreen mode.
+  if (!enter_fullscreen && forced_fullscreen_) {
```

Comment:
Yes and it works as expected in this PR.
If the window is already in fullscreen when the html context wants fullscreen we just allow the html to go fullscreen without changing the window state. And when the html context wants to exit fullscreen we check to see if the window state was changed on entering html fullscreen if not we just leave it as it is.